Documentation/guides/tracing.md

```diff
@@ -67,19 +61,33 @@ dbug: dotnet-dsrouter[0]
       Waiting for new ipc connection at endpoint "dotnet-diagnostic-dsrouter-21352".
 ```
 
-### For Android devices
+Use `dotnet-dsrouter android` instead, if you are testing on an
+Android device. The `adb reverse` command mentioned above is required
+for physical devices.
+
+### Configure the device so that the profiled app suspends until tracing utility connects
 
-For profiling an Android application running on an Android device:
+Note the log message that `dotnet-dsrouter` prints that mentions
+`$DOTNET_DiagnosticPorts`. `$DOTNET_DiagnosticPorts` is an environment
+variable that could be defined in an `@(AndroidEnvironment)`, but it
+is simpler to use the `debug.mono.profile` Android system property.
+Android system properties can be used without rebuilding the app.
 
-```
-$ dotnet-dsrouter server-server -tcps 127.0.0.1:9001 --verbose debug
+Based on the value `dotnet-dsrouter` logs for
+`DOTNET_DiagnosticPorts`, choose one of:
```

Comment:
These two paragraphs should be "inverted", so that the selection criteria is topmost, rather than "nested in a comment":

> For emulators, `DOTNET_DiagnosticPorts` should specify an IP address of `10.0.2.2` ( [ed question: will this ever *not* be the case?  when?  Is there some way to verify this? ] ), e.g.
>
>      adb shell setprop debug.mono.profile '10.0.2.2:9000,suspend,connect'
>
> For devices, `DOTNET_DiagnosticPorts` should specify an IP address of `127.0.0.1`, and the port number should be the \[port used used with `adb reverse`\]\(# TODO: provide an intra-page link to the "Set up reverse port forwarding" section \), e.g.
>
>     $ adb shell setprop debug.mono.profile '127.0.0.1:43399,suspend,connect'